hooks/pre/30-upgrade-postgresql.rb

```diff
@@ -0,0 +1,18 @@
+def postgresql_10_upgrade
+  execute('foreman-maintain service start --only=postgresql')
+  (name, owner, enconding, collate, ctype, privileges) = %x{runuser postgres -c 'psql -lt | grep -E "^\s+postgres"'}.chomp.split('|').map(&:strip)
+  execute('foreman-maintain service stop')
+  execute('yum -y install rh-postgresql10-postgresql-server')
+  execute(%Q{scl enable rh-postgresql10 "PGSETUP_INITDB_OPTIONS='--lc-collate=#{collate} --lc-ctype=#{ctype} --locale=#{collate}' postgresql-setup --upgrade"})
+  execute('yum -y remove postgresql postgresql-server')
+  execute('rm -f /etc/systemd/system/postgresql.service')
+  execute('yum -y install rh-postgresql10-syspaths')
+end
+
+if local_postgresql? && facts[:os][:release][:major] == '7' && module_enabled?('foreman')
```

Comment:
Should this check for os family being RedHat? Not that I envision us to run on Debian 7, but just to be thorough… :)